lib/util.js

```diff
@@ -420,7 +420,15 @@ function formatValue(ctx, value, recurseTimes, ln) {
   // Provide a hook for user-specified inspect functions.
   // Check that value is an object with an inspect function on it
   if (ctx.customInspect) {
-    const maybeCustomInspect = value[customInspectSymbol] || value.inspect;
+    let deprecated;
+    if (value.inspect && value.inspect !== exports.inspect) {
+      deprecated = deprecate(
+        value.inspect,
+        'Custom inspection function on Objects via .inspect() is deprecated',
+        'DEP0079'
+      );
+    }
```

Comment:
I don’t think we’d want to generate a deprecated function every time `value.inspect` is present. I’d prefer at least checking `value[customInspectSymbol]` first, before going through this trouble. (And if you want to go further, maybe a WeakMap to keep the generated closures cached for performance – but that might not be *that* important for deprecated features…)

Reply:
It should also only trigger in case it is a function. Right now it would always trigger.

Reply:
OK, took care of the "it will generate a deprecated function every time" issue.

Reply:
For simplicity, I'd fire the warning here, and not delegate it to the actual call, or just set a flag and fire next to the call. That saves the closure creation.

Reply:
@BridgeAR Yeah, definitely should check for `typeof` there. Will add that in. Thanks!

Reply:
`typeof` check added along with test that exercises the additional code branch.